sql/repl_failsafe.cc

```diff
@@ -186,11 +188,15 @@ static my_bool show_slave_hosts_callback(THD *thd, Protocol *protocol)
     {
       gtid_state_from_binlog_pos(si->gtid_state_sent.log_file, (uint32)si->gtid_state_sent.log_pos, gtid_sent);
       gtid_state_from_binlog_pos(si->gtid_state_ack.log_file, (uint32)si->gtid_state_ack.log_pos, gtid_ack);
+      sync_str= (char *)sync_str_opt[1];
```

Comment:
Let's use a ternary operator here to set `sync_str`. Then a few other general points for this code block:

1. `gtid_state_sent` should always be output, even for async slaves, so that should be out of the `if` conditional.
2. The `if` conditional should be based on `thd->semi_sync_slave`, as the primary can support multiple slaves of different connection types.